app/customer.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+from math import sqrt
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: dict,
+            money: int,
+            car: Car
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = car
+        self.shop_trips = {}
+
+    def go_to_the_shop(self, shops: list[Shop]) -> Shop:
+        self.amount_of_money()
+        for shop in shops:
+            self.get_shop_trip_cost(shop)
+
+        shop = self.drive_to_beneficial_shop()
+        return shop
+
+    def drive_home(self) -> None:
+        self.money -= (min(self.shop_trips))
+        print(f"{self.name} rides home")
+        print(f"{self.name} now has {self.money} dollars\n")
+
+    def amount_of_money(self) -> None:
+        print(f"{self.name} has {self.money} dollars")
+
+    def drive_to_beneficial_shop(self) -> Shop | None:
+        if min(self.shop_trips) > self.money:
+            print(f"{self.name} doesn't have enough money "
+                  f"to make a purchase in any shop")
+            return None
+
+        cheapest_shop = self.shop_trips[min(self.shop_trips)]
+        print(f"{self.name} rides to {cheapest_shop.name}\n")
+        return cheapest_shop
+
+    def get_shop_trip_cost(self, shop: Shop) -> None:
+        distance_price = self.get_distance_price(shop.location)
+        products_price = self.get_product_price(shop.products)
+        cost_of_trip = round(distance_price + products_price, 2)
+        print(f"{self.name}'s trip to the {shop.name} costs {cost_of_trip}")
+        self.shop_trips[cost_of_trip] = shop
+
+    def get_distance_price(self, shop_location: dict) -> float:
```

Comment:
this would be rather a Car method - not a customer method. it doesn't depend on a customer in any way so it would be logical to move it out. Your customer already has a lot of methods and having one class that does it all is not always a good idea

Reply:
I also had doubts about this method. Now it’s clear :) Thanks!